python/tests/integration/arcticdb/test_storage_lock.py

```diff
@@ -0,0 +1,49 @@
+import pandas as pd
+import numpy as np
+import pytest
+
+from arcticdb_ext.tools import ReliableStorageLock, ReliableStorageLockManager
+from tests.util.mark import PERSISTENT_STORAGE_TESTS_ENABLED, REAL_S3_TESTS_MARK
+
+from multiprocessing import Process, set_start_method
+set_start_method("fork") # Okay to fork an S3 lib
+import time
+
+from arcticdb.util.test import assert_frame_equal
+
+
+one_sec = 1_000_000_000
+
+
+def slow_increment_task(lib, symbol, sleep_time, lock_manager, lock):
+    lock_manager.take_lock_guard(lock)
+    df = lib.read(symbol).data
+    df["col"][0] = df["col"][0] + 1
+    time.sleep(sleep_time)
+    lib.write(symbol, df)
+    lock_manager.free_lock_guard()
+
+
+@pytest.mark.parametrize("num_processes,max_sleep", [(100, 1), (5, 20)])
+@REAL_S3_TESTS_MARK
+def test_many_increments(real_s3_version_store, num_processes, max_sleep):
+    lib = real_s3_version_store
+    init_df = pd.DataFrame({"col": [0]})
+    symbol = "counter"
+    lib.write(symbol, init_df)
+    lock = ReliableStorageLock("test_lock", lib._library, 10*one_sec)
+    lock_manager = ReliableStorageLockManager()
+
+    processes = [
+        Process(target=slow_increment_task, args=(lib, symbol, 0 if i%2==0 else max_sleep, lock_manager, lock))
+        for i in range(num_processes)
+    ]
+    for p in processes:
+        p.start()
+
+    for p in processes:
+        p.join()
+
+    read_df = lib.read(symbol).data
+    expected_df = pd.DataFrame({"col": [num_processes]})
+    assert_frame_equal(read_df, expected_df)
```

Comment:
Also assert on the version number after all these writes?

Reply:
I've noticed when running these that the persistent storage might not get cleaned up after a previous test run especially when running locally. So I think it might be possible to do these writes on a library which already has the `counter` written with some arbitrary version. I.e. it's possible we get a version 200 because the test was run before.

I think the counter being equal to 100 is essentially the same assertions as writing 100 versions, so there isn't much benefit and it makes local testing slightly more annoying. This ifine for the CI I think. What do you think?

Reply:
Would be simple to do a force delete symbol at the start of the test?

Reply:
Ah didn't know about the `force_delete_symbol` API. Done, thanks!